content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,430 @@
+
+Prisma 4 and lower:
+
+```js
+prisma.post.create({
+  data: { tags: 'databases' },
+})
+
+prisma.post.findMany({
+  where: { tags: 'databases' },
+})
+```
+
+Prisma 5:
+
+```js highlight=3,9;edit
+prisma.post.create({
+  data: {
+    tags: ['databases'],
+  },
+})
+
+prisma.post.findMany({
+  where: {
+    tags: ['databases'],
+  },
+})
+```
+
+### Prisma Client embedded SQLite version updated
+
+With Prisma version 5.0.0, we have upgraded the embedded version of SQLite from `3.35.4` to `3.41.2`. We did not see any breaking changes and don't anticipate any changes needed in user projects, but if you are using SQLite, make sure to check [the SQLite changelog](https://www.sqlite.org/changes.html)
+
+### Removal of <inlinecode>jsonProtocol</inlinecode> preview feature
+
+The `jsonProtocol` preview feature is now generally available! When upgrading to Prisma 5, make sure to remove `jsonProtocol` from your preview features, if added.
+
+Prisma 4 and lower:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["jsonProtocol", "tracing"]
+}
+```
+
+Prisma 5:
+
+```prisma highlight=3;delete
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["tracing"]
+}
+```
+
+In addition, several error messages have been improved. The following error message in Prisma 4 and below
+
+```text
+Argument bool: Got invalid value 'true' on prisma.createOneD. Provided String, expected Boolean.
+Argument byteA: Got invalid value 'hello prisma ⚡️🚀' on prisma.createOneD. Provided String, expected Bytes.
+Argument xml: Got invalid value 123 on prisma.createOneD. Provided Int, expected String.
+```
+
+becomes the following in Prisma 5:
+
+```text
+Argument `bool`: Invalid value provided. Expected Boolean, provided String.
+```
+
+### <inlinecode>cockroachdb</inlinecode> provider is now required when connecting to a CockroachDB database
+
+### Removal of <inlinecode>runtime/index.js</inlinecode> from generated client
+
+#### Using public APIs from <inlinecode>runtime/index.js</inlinecode>
+
+Importing from `@prisma/client/runtime` is no longer available in Prisma 5. If you are using public APIs available in this namespace you can instead import `Prisma` and access them. For example:
+
+```js
+import { Decimal } from '@prisma/client/runtime'
+```
+
+will need to be changed to
+
+```js
+import { Prisma } from '@prisma/client'
+Prisma.Decimal
+```
+
+#### Using private APIs for a specific runtime
+
+Instead of importing private APIs from the top-level `@prisma/client/runtime` you will need to import from the runtime specific to your database engine. For example:
+
+```text
+@prisma/client/runtime/data-proxy
+```
+
+## Other changes
+
+The following changes may cause an application to initially throw an error message after upgrading to Prisma 5. Fortunately, they are easy to solve, as the underlying functionality has been removed for a while or the change is a simple string replace.
```

Comment:
Aren't these _also_ breaking changes then?

I like the current headlines better than "major" and "minor", but won't a user rightfully be confused that something that breaks things is not identified as "breaking changes".

"Main" and "Other" could work maybe.

Reply:
Main and Other sounds good to me. Updated.